server/protocols/webadmin/webadmin-mailrepository/src/main/java/org/apache/james/webadmin/routes/MailRepositoriesRoutes.java

```diff
@@ -399,4 +418,20 @@ private void enforceActionParameter(Request request) {
     private String decodedRepositoryUrl(Request request) throws UnsupportedEncodingException {
         return URLDecoder.decode(request.params("encodedUrl"), StandardCharsets.UTF_8.displayName());
     }
+
+    private List<AdditionalField> extractAdditionalFields(String additionalFieldsParam) throws IllegalArgumentException {
+        List<AdditionalField> additionalFields = new ArrayList<>();
+
+        for (String requestedField : Splitter.on(',').trimResults().omitEmptyStrings().split(additionalFieldsParam)) {
+            Optional<AdditionalField> correspondingField = AdditionalField.find(requestedField);
+            if (correspondingField.isPresent()) {
```

Comment:
a isPresent in a if is generally a design smell. You should use optional as a monad...

For the record here:

```
additionalFields.add(correspondingField
    .orElseThrow(() -> new IllegalArgumentException(requestedField)));
```

Does the same thing than your code